src/filters/date/time/time.ts

```diff
@@ -25,43 +25,25 @@ const prepositionToTrad: { [key: string]: string } = {
 
 export type SupportedTimeTypes = Date | string | undefined;
 export interface MTimeRange<T extends SupportedTimeTypes = string> {
-    from: T;
-    to: T;
+    from?: T;
+    to?: T;
 }
 
-export const timeFilter: (time: SupportedTimeTypes, options?: TimeFilterOptions) => string = (time, options: TimeFilterOptions = { preposition: TimeFilterPrepositions.None }) => {
-    if (!time) {
-        return '';
-    }
-
-    const now: Date = new Date();
-
-    let datePart: string = '';
-    let timePart: string = '';
-    if (time instanceof Date) {
-        datePart = `${time.getFullYear()}-${(time.getMonth() + 1).toString().padStart(2, '0')}-${time.getDate().toString().padStart(2, '0')}`;
-        timePart = `${time.getHours().toString().padStart(2, '0')}:${time.getMinutes().toString().padStart(2, '0')}`;
-    } else if (typeof time === 'string') {
-        datePart = `${now.getFullYear()}-${(now.getMonth() + 1).toString().padStart(2, '0')}-${now.getDate().toString().padStart(2, '0')}`;
-        timePart = !time.includes(':') ? `${time.padStart(2, '0')}:00` : time;
-    }
-
+function getLocalDateTime(time: SupportedTimeTypes): Date | undefined {
     try {
-        const date: Date = new Date(`${datePart}T${timePart}`);
-        const intlOptions: Intl.DateTimeFormatOptions = {
-            hour: 'numeric',
-            minute: date.getMinutes() !== 0 ? 'numeric' : undefined
```

Comment:
Est-ce possible de remplacer les « 00 h » par « 0 h » en français et « 04 h » par « 4 h » ? 
![image](https://user-images.githubusercontent.com/35844644/60898960-acf2b900-a237-11e9-9669-f9d1d78018f5.png)

Reply:
C'est pas built-in dans Intl et mois je suis pas chaud à l'idée d'overrider Intl pour le moment.  J'aimerais qu'on en reparle en gang avant de procéder.

Du code du genre if (locale === en_CA) => Faire quelque chose
if (locale === fr_CA) => Faire autre chose

C'est coûteux à entretenir.

Sinon on pourrait p-e trouver une librarie d'interntionalisation qui fait déjà tout ce qu'on veut?

@chuckmah @vidal7

Reply:
Je suis d’accord avec toi qu’idéalement la librairie devrait faire ce qu’on veut idéalement.

Reply:
Si on lit sur le bureau de la traduction, en anglais, [deux options sont acceptées](http://www.btb.termiumplus.gc.ca/wrtps-srch?lang=eng&srchtxt=colons&cur=4&nmbr=4&lettr=indx_catlog_t&page=9lMKm-R0uJrs.html) pour l'affichage des heures piles. Soit 5:00 p.m. ou 5 p.m. On avait opté pour 5:00 p.m. qu'on trouvait plus clair, surtout dans les intervalles comme « 9:30 a.m. - 12 p.m. ».  La librairie intl a choisi de ne pas les mettre. Comme les deux sont acceptés selon le BTB, c'est correct pour ce point.

Pour le « 0 » en avant d'une heure en français comme dans « 04 h 30 » ou « 00 h », ça c'est un peu plus chiant, il va falloir trouver une solution. Mathieu a saisi un [bogue](https://bugs.chromium.org/p/chromium/issues/detail?id=982778) dans chronuim. On peut attendre un peu et voir la réaction.

Reply:
En fait mettre le 5:00 rien m'empêche de le faire.

Reply:
Mais dans ce cas il y aura un breaking change

Reply:
Si c'est possible ce serait bien de le faire alors :)